### DISCUSSION TITLE
Download files where content disposition is set

### DISCUSSION THREAD

**Discussion by mlambley** [Kong/insomnia#3948] [Ideas] [answered]
Hi Greg and team,

I was wondering if, where the site serves a content disposition, if a file save dialog could appear and allow the user to download it to their device. I'm testing code with:
```php
'Content-Type', 'application/vnd.ms-excel'
'Content-Disposition', 'attachment;filename="My File.xlsx"'
```

The response is basically:
![image](https://user-images.githubusercontent.com/17264636/130882700-e5595143-2311-4921-873b-12acc83b10e6.png)

Right now I have to create special routes which serve it up over a GET request with no authentication, and then open it up in a browser in order to debug it. Is there any chance I can download the files via Insomnia?

Thank you.

**Comment by dimitropoulos** [comment]
Would you, perchance, be able to craft a mockbin.org link that replicates the problem?  From there, we can start talking about what a "download file" functionality in insomnia might look like.

**Comment by dimitropoulos** [comment]
Just to verify, your use-case would be satisfied if when the request is made, there is a prompt or button that appears or _something_ that you have to click to actually download the file, right?  I ask because I'm not sure we'd see it as a security risk or not for insomnia to be automatically downloading files.

I imagine we'd probably get a regular opperating-system file-save prompt to appear that would allow you to choose where the file is downloaded.

**Comment by mlambley** [comment]
I just had a look at mockbin and can't see how to replicate a binary file.

The use case would be:
1) Insomnia gets a response with Content-Disposition set
2) User sees a file download dialog. I've attached a screenshot of an example.
3) The file dialog gives them the option to save to a particular folder, or to cancel and not download it.

![image](https://user-images.githubusercontent.com/17264636/131587140-5e7699d1-eefc-4659-b0bd-67dae8788021.png)

**Reply by shailendra-burman** [reply]
Any resolution here?

**Comment by seb-pereira** [comment, marked as answer]
To get the file, you could select `raw > Export raw response` and optionally set the appropriate file extension.

<img width="439" alt="image" src="https://github.com/user-attachments/assets/f15a534c-22f5-46a4-81ba-20e4a7920c2a" />

I do this with zip files and it works fine (version 10.3.1)

**Comment by kevingangi** [comment]
This does not work in v12.5.0.  It appears to be a regression of some kind.

File exports as 0 length file.
